tests/posix/common/src/pthread.c

```diff
@@ -232,19 +269,49 @@ void test_pthread(void)
 		sem_wait(&main_sem);
 	}
 
-	/*TESTPOINT: Check if barrier test passes*/
+	/* TESTPOINT: Check if barrier test passes */
 	zassert_false(barrier_failed, "Barrier test failed");
 
-	for (i = 0; i < N_THR; i++) {
+	for (i = 0; i < N_THR_E; i++) {
 		pthread_join(newthread[i], &retval);
 	}
 
 	printk("Barrier test OK\n");
 }
 
-void test_main(void)
+void test_posix_pthread_termination(void)
```

Comment:
Will be adding tests from previous pthread_join test here, modifying them to cover remaining exit scenarios. This is currently the pthread_cancel test, which actually covers a good amount of pthread_join cases too.